00_Project/sos/SERVICES/sos/sos_interface.h

```diff
@@ -0,0 +1,138 @@
+ * Created: 5/29/2023 7:44:00 PM
+ *  Author: sharpel
+ */ 
+
+
+#ifndef SOS_INTERFACE_H_
+#define SOS_INTERFACE_H_
+
+#define MAX_TASKS  10
+
+// assuming that lower the priority number having higher priority 
+typedef enum
+{
+  PRIORITY_ZERO = 0,	
+  PRIORITY_ONE,
+  PRIORITY_TWO,
+  PRIORITY_THREE,
+  PRIORITY_FOUR,
+  PRIORITY_FIVE,
+  PRIORITY_SIX,
+  PRIORITY_SEVEN,
+  PRIORITY_EIGHT,
+  PRIORITY_NINE,
+  PRIORITY_TOTAL,	
+}enu_task_priority_id_t; 
+typedef struct  
+{
+	Uint16_t taskPeriodicity;
+	void (*ptr_task_ref)(void*);
+	void *ptr_task_arg;
+	
+}str_tasks_t;
+
+
+typedef enum
+{
+	SOS_STATUS_SUCCESS,
+	SOS_STATUS_INVALID_STATE,
+	SOS_NULL_ARGS,
+	SOS_TASK_PERIODICITY_UNKNOWN,
+	SOS_TASK_DUBLICATED_PIRIORITY,
+	SOS_TASK_NOT_FOUND,
+	SOS_TASK_PRIRORITY_ERROR,
+}enu_system_status_t;
+
+typedef enum
+{
+	IDLE,
+	RUNNING,
+	RUNNING_DISABLED,
+}enu_run_state_t;
+
+
+
+/**
+ * @brief                                               :   Function Used to Initialize sos
+ * @name                                                :   sos_init
+ * @param[in,out]                                       :   None
+ * @param[in]                                           :   None   
+ * @return      SOS_STATUS_SUCCESS                      :   In case of Successful Operation     
+ *              SOS_STATUS_INVALID_STATE                :   In case of re-initialization    
+ *             
+ */
+enu_system_status_t sos_init(void);
+
+/**
+ * @brief                                               :   Function Used to de-Initialize sos
+ * @name                                                :   sos_deinit
+ * @param[in,out]                                       :   None
+ * @param[in]                                           :   None   
+ * @return      SOS_STATUS_SUCCESS                      :   In case of Successful Operation     
+ *              SOS_STATUS_INVALID_STATE                :   In case of duplicated deinitialization or the sos not initialized before   
+ *             
+ */
+enu_system_status_t sos_deinit(void);
+
+
+/**
+ * @brief                                               :   Function Used to Create task and store it in database 
+ * @name                                                :   sos_create_task
+ * @param[in,out]                                       :   None
+ * @param[in]                                           :   enu_task_priority_id_t
+                                                            str_tasks_config_t  
+ * @return      SOS_STATUS_SUCCESS                      :   In case of Successful Operation     
+ *              SOS_NULL_ARGS                           :   In case of Null pointer arguments
+                SOS_TASK_PERIODICITY_UNKNOWN            :   In case of undefined periodicity
+                SOS_TASK_DUBLICATED_PIRIORITY           :   In case of duplicated priority
+ *             
+ */
+enu_system_status_t sos_create_task(enu_task_priority_id_t enu_task_priority_id , str_tasks_t *str_tasks_config);
```

Comment:
Rename the `str_tasks_config `to become `str_tasks_parameters`

Reply:
Done